### PR TITLE
Refactoring spirv_generator.py

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
Code simplification.

### CODE REVIEW

File: numba_dppy/spirv_generator.py [hunk 1]
```diff
@@ -42,61 +42,49 @@ def check_call(*args, **kwargs):
...
-                # "--no-header",
-                # "--raw-id",
-                # "--offsets",
-                "-o",
-                opath,
-                ipath,
-            ]
-        )
+        flags = []
+        # flags.append("--no-indent")
```
Comment (diptorupd) [inline numba_dppy/spirv_generator.py:46]: Let us get rid of the commented out lines.

File: numba_dppy/spirv_generator.py [hunk 2]
```diff
@@ -42,61 +42,49 @@ def check_call(*args, **kwargs):
...
-                # "--inline-entry-points-exhaustive",
-                # "--flatten-decorations",
-                # "--compact-ids",
-                "-o",
-                opath,
-                ipath,
-            ]
-        )
+        flags = []
+        # flags.append("--strip-debug")
```
Comment (diptorupd) [inline numba_dppy/spirv_generator.py:57]: same here.

File: numba_dppy/spirv_generator.py [hunk 3]
```diff
@@ -42,61 +42,49 @@ def check_call(*args, **kwargs):
...
 
+        ipath_optimized = ipath + ".bc"
+
         # Get optimization level from NUMBA_OPT
-        opt_level_option = f"-O{config.OPT}"
+        opt_flags = [f"-O{config.OPT}"]
 
-        check_call(["opt", opt_level_option, "-o", ipath + ".bc", ipath])
-        check_call(["llvm-spirv", "-o", opath, ipath + ".bc"])
+        check_call(["opt", *opt_flags, "-o", ipath_optimized, ipath])
```
Comment (diptorupd) [inline numba_dppy/spirv_generator.py:79]: Ideally we should use `llvm-as` instead of `opt`.

File: numba_dppy/spirv_generator.py (hunk 1 above)
Comment (PokhodenkoSA) [reply numba_dppy/spirv_generator.py:46]: So I need another way to provide info for which flags are possible.

**Pull request closed without merging**